app/server/identity/identityMiddleware.ts

```diff
@@ -27,18 +32,33 @@ export const withIdentity: express.RequestHandler = (
 
   if (cookies[GU_U] == null || cookies[SC_GU_U] == null) {
     log.info("Not logged in.");
-    const returnUrl = url.format({
-      protocol: "https",
-      host: req.get("host"),
-      pathname: req.originalUrl
-    });
+
     // somehow the redirect url is automatically encoded
     res.redirect(
       `https://profile.${conf.DOMAIN}/signin?returnUrl=${returnUrl}`
     );
     return;
   }
-
+  try {
+    const now = new Date().getTime();
+    const cookieString = atob(cookies[SC_GU_U].split(".")[0]);
+    const cookieData = JSON.parse(cookieString);
+    const expiry = cookieData[1];
+    const remaining = expiry - now;
+    if (remaining < 60) {
+      log.info("Insufficient time left in session.");
+      res.redirect(
+        `https://profile.${conf.DOMAIN}/signin?returnUrl=${returnUrl}`
```

Comment:
probably want to move this full redirectURL into constant above, rather than repeating it twice (line 51 & 58)